docarray/array/mixins/parallel.py

```diff
@@ -60,11 +60,8 @@ def apply(self: 'T', *args, **kwargs) -> 'T':
         # noqa: DAR201
         :return: a new :class:`DocumentArray`
         """
-        from ... import DocumentArray
-
-        new_da = DocumentArray(self.map(*args, **kwargs))
-        self.clear()
-        self.extend(new_da)
+        for doc in self.map(*args, **kwargs):
```

Comment:
use batch_map would be efficient for some backends?

Reply:
hmm, interesting thought, that could be true. especially when the afterward `da[d.id] = d` is IO bounded https://docarray.jina.ai/fundamentals/documentarray/parallelization/#use-map-batch-to-overlap-cpu-and-network-time

@alaeddine-13 you think?

Reply:
then it's better to use it for `apply_batch` not `apply`.
I have applied the same logic for `apply_batch`, can you check ?